src/amo/api/index.js

```diff
@@ -1,14 +1,31 @@
 import { callApi } from 'core/api';
-import log from 'core/logger';
 
-export function postRating({ rating, apiState, addonId, versionId }) {
-  const postData = { rating, version: versionId };
-  log.debug('about to post add-on rating with', postData);
-  return callApi({
-    endpoint: `addons/addon/${addonId}/reviews`,
-    body: postData,
-    method: 'post',
-    auth: true,
-    state: apiState,
-  });
+/*
+ * POST/PATCH an add-on review using the API.
+ */
+export function submitReview({
+  rating, apiState, addonId, versionId, body, reviewId,
+}) {
+  const data = { rating, version: versionId, body };
+  let method = 'POST';
+  let endpoint = `addons/addon/${addonId}/reviews`;
+
+  return new Promise(
+    (resolve) => {
+      if (!addonId) {
+        throw new Error('addonId is required to build the endpoint');
+      }
+      if (reviewId) {
+        endpoint = `${endpoint}/${reviewId}`;
+        method = 'PATCH';
+      }
+      resolve();
+    })
+    .then(() => callApi({
```

Comment:
Sure, it could. I like to minimize the use of custom Promise objects as much as possible since it's a common mistake to refactor some code that used to use `resolve()` and break it when you switch it to `then()`. I can merge the two calls together though.